internal/io/io.go

```diff
@@ -4,24 +4,27 @@ import (
 	"io/ioutil"
 
 	"github.com/fusor/cpma/internal/io/sftpclient"
-	"github.com/sirupsen/logrus"
 )
 
 var GetFile = fetchFile
 
 // Fetch first tries to retrieve file from local disk (outputDir/<Hostname>/).
 // If it fails then connects to Hostname to retrieve file and stores it locally
 // To force a network connection remove outputDir/... prior to exec.
-func fetchFile(host, src, target string) []byte {
+func fetchFile(host, src, target string) ([]byte, error) {
 	f, err := ioutil.ReadFile(target)
+	if err == nil {
+		return f, nil
+	}
+
+	err = sftpclient.Fetch(host, src, target)
```

Comment:
I would add comments to this method. It took me a little bit to see that you are reading the file from the target but if nothing is there you will attempt to fetch the file from the host. A comment above the Fetch would help:
```golang
// Cached file does not exist, fetch from host
err = sftpclient.Fetch(host, src, target)
...
```